IaC/scripts/create-cluster-env.bash

```diff
@@ -0,0 +1,398 @@
+#!/bin/bash
+## TODO:
+##   Put descriptive echo's in Color
+##   Cleanup script [sem-done]
+## Qurey
+##   Zone and region needed to be selected for both smoker and ngsa
+
+# Sane prog env:
+set -o errexit -o pipefail -o noclobber # -o nounset
+aze() { echo "executing: az $@";az "$@"; }
+dockere() { echo "executing: docker $@";docker "$@"; }
+kubectle() { echo "executing: kubectl $@";kubectl "$@"; }
+# Functions
+usage(){
+    echo """
+    $0 --ngsa-prefix basename123 [Optional Args/Flags]
+    $0 -s azure-subs -n basename123 [Optional Args/Flags]
+
+Required args:
+    -n | --ngsa-prefix BASE_NAME    This will be the NGSA prefix for all resources
+                                    Do not include punctuation - only use a-z and 0-9
+                                    must be at least 5 characters long
+                                    must start with a-z (only lowercase)
+Optional args:
+    -s | --subscription AZ_SUB      Azure Subscription Name or ID
+    -e | --env ENVIRONMENT          Environemnt Type. Default: dev (See README.md for other values)
+    -d | --domain DOMAIN_NAME       Registered Domain Name. Default: nip.io (Requires --email)
+    -m | --email EMAIL_DOMAIN       Required Email if a '--domain' is given
+    -l | --location LOCATION        Location where the resources will be created. Default: westus2
+    -k | --k8s-ver K8S_VERSION      Kubernetes version used. Default: 1.18.8
+                                    Use 'az aks get-versions -l westus2 -o table' to get supported versions
+    -c | --node-count NODE_COUNT    Cluster Node Count. Default: 3
+    -r | --dns-rg DNS_RG            DNS Resource group name. Default: dns-rg
+    -i | --cosmos-key COSMOS_KEY    
+    -u | --cosmos-url COSMOS_URL    In case users want to use their own CosmosDBBoth Key and URL are empty by default. 
+Optional Flag:
```

Comment:
Are the flags below optional as well? or the ones above should be renamed as required flags?

Reply:
Yes All flags and arguments are optional except `-n | --ngsa-prefix`.